### PR TITLE
upgrade corretto major version to 15

### PR DESCRIPTION
**Pull request by niels1voo**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [ ] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/corretto.rb [hunk 1]
```diff
@@ -1,10 +1,11 @@
...
-  sha256 "d87c677c3f2d5b323112aa8e028cd4dbaad0e37fb378ffac87f7e447996476cb"
+  version "15.0.0.36.1"
+  sha256 "80e0c93299bfd36bd325d9b40aab8531a5d1d860338a9d54e574180f02a4502d"
 
   url "https://corretto.aws/downloads/resources/#{version.sub(/-\d+/, "")}/amazon-corretto-#{version}-macosx-x64.pkg"
-  appcast "https://www.macupdater.net/cgi-bin/check_urls/check_url_redirect.cgi?url=https://corretto.aws/downloads/latest/amazon-corretto-11-x64-macos-jdk.pkg"
-  name "Amazon Corretto"
+  appcast "https://www.macupdater.net/cgi-bin/check_urls/check_url_redirect.cgi?url=https://corretto.aws/downloads/latest/amazon-corretto-#{version.major}-x64-macos-jdk.pkg"
+  name "AWS Corretto JDK"
+  desc "AWS Corretto JDK"
```
Comment (reitermarkus) [inline Casks/corretto.rb:8]: Please add a better description.

**Comment by miccal** [comment]
@niels1voo the main `homebrew-cask` repo is for the latest versions, so please instead just update the existing `corretto` Cask to version 15 and remove the `corretto15` Cask you have added.

File: Casks/corretto.rb (hunk 1 above)
Comment (niels1voo) [reply Casks/corretto.rb]: updating it.  strange that it passes audit locally but p/r process fails it for line length at two different thresholds.

**Comment by miccal** [comment]
Thank you @niels1voo.

**Review by miccal — APPROVED** [review]

**Comment by y-higuchi** [comment]
@miccal Should `corretto` be point to a preview version?

The official corretto page only list corretto 11 and 8 as of right now https://aws.amazon.com/corretto/
and corretto 15 page describe itself as Developer Preview.
https://docs.aws.amazon.com/corretto/latest/corretto-15-ug/doc-history.html

Shouldn't `corretto` be pointing to the latest stable version 11?

**Comment by miccal** [comment]
@y-higuchi I am not really sure I am afraid -- I just assumed that version 15 was the latest version.

Ping @vitorgalvao and @reitermarkus for advice here please.

**Comment by vitorgalvao** [comment]
> The official corretto page only list corretto 11 and 8 as of right now https://aws.amazon.com/corretto/

I’m seeing the three versions.

image

> Shouldn't `corretto` be pointing to the latest stable version 11?

Yes, we should be pointing to whatever is the latest stable version, which to be frank isn’t really clear.

**Comment by y-higuchi** [comment]
You're right, I didn't see it when I looked at it before so may be they are updating the pages soon?

On Java 11 was release, it looks like it became officially GA roughly 1 month after initial developer preview, 
https://docs.aws.amazon.com/corretto/latest/corretto-11-ug/doc-history.html

It's roughly 1 month from initial Java 15 developer preview from the document history so timing does look similar.
Maybe corretto 15 will officially become latest stable in couple days. 

So pointing corretto to 15 and accepting corretto11 might be the reasonable way forward

**Pull request merged**